### ISSUE TITLE
How valuable would table storage perister for sagas that don't need the lookup documents?

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus.Persistence.AzureStorage#134]
The core will soon enable persister use a single id for lookups

https://github.com/Particular/NServiceBus/pull/4496

this means that you would not need the lookup document. 

How much would that mode improve the table storage persister?

**Comment by Scooletz** [comment]
That's true. Having a custom saga id generator would simplify storage for newly created types of sagas. Although for historically existing this is not the case. To maintain the backward compatibility, secondary indexes in ASP are the ones based on the correlation property. Saga data are stored primarily by the saga identifier. If we changed the id generation we'd need:
- a flag for persistence that it should not create secondary indexes 
- a flag for persistence that it should correlate using the id generator
- a way of do the full switch for all the scaled out endpoints (downtime & upgrade is the only idea that comes to my mind now), where no sagas are created

This does not solve messages coming with the correlation id that need to be correlated to old sagas by its old secondary index.

To sum up. If we're discussing a brand new deployment that's easy. If we're considering a migration scenario, it's not that easy, if possible at all. The easiest way that comes to my mind is a new persister API that could be easily and effectively implemented, a.k.a. `Simple saga`.

Ping @andreasohlund

**Comment by andreasohlund** [comment]
>To sum up. If we're discussing a brand new deployment that's easy. If we're considering a migration scenario, it's not that easy, if possible at all. 

I always had a brand new persister in mind so 👍

Would a blob storage persister be relevant?

**Comment by Scooletz** [comment]
A blob storage persister would be beneficial for sagas bigger than:

- max size of a table entity **1 MB**
- max number of properties in a table entity 	**252**

According to my understanding, we guide to not create that big sagas. If that's true, I see no value in creating blob storage persister.

ping @andreasohlund

**Comment by andreasohlund** [comment]
What about storage costs, Performance, CDN-ability" etc?

**Comment by Scooletz** [comment]
As far as I know all these makes no real difference. @Particular/azure-maintainers could you chime in and validate my assumptions?

**Comment by SeanFeldman** [comment]
It's the same service behind the scenes, so doubt you'll see such a huge difference.
CDN shouldn't be tied into that as it's not part of the Storage.

**Comment by Scooletz** [comment]
If the above answers your questions, could you close it @andreasohlund
